packages/superset-ui-control-utils/src/types.ts

```diff
@@ -52,13 +52,8 @@ export interface DatasourceMeta {
 
 export interface ControlPanelState {
   form_data: QueryFormData;
-  datasource?: DatasourceMeta | null;
-  options?: ColumnMeta[];
-  controls?: {
-    comparison_type?: {
-      value: string;
-    };
-  };
+  datasource: DatasourceMeta | null;
+  controls: ControlStateMapping;
```

Comment:
This field is now required and also less strict about its value.

Reply:
Yes, it is a required field based on source code in superset-frontend. It's basically a mapping of the states of all current controls.

Reply:
cool